pkg/controller/registry/reconciler.go

```diff
@@ -0,0 +1,360 @@
+	return map[string]string{
+		"olm.catalogSource":            s.GetName(),
+		"olm.configMapResourceVersion": s.Status.ConfigMapResource.ResourceVersion,
+	}
+}
+
+func (s *catalogSourceDecorator) ConfigMapChanges(configMap *v1.ConfigMap) bool {
+	if s.Status.ConfigMapResource == nil {
+		return true
+	}
+	if s.Status.ConfigMapResource.ResourceVersion == configMap.GetResourceVersion() {
+		return false
+	}
+	return true
+}
+
+func (s *catalogSourceDecorator) Service() *v1.Service {
+	svc := &v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.GetName(),
+			Namespace: s.GetNamespace(),
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				{
+					Name:       "grpc",
+					Port:       50051,
+					TargetPort: intstr.FromInt(50051),
+				},
+			},
+			Selector: s.Labels(),
+		},
+	}
+	ownerutil.AddOwner(svc, s.CatalogSource, false, false)
+	return svc
+}
+
+func (s *catalogSourceDecorator) Pod(image string) *v1.Pod {
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: s.GetName() + "-",
+			Namespace:    s.GetNamespace(),
+			Labels:       s.Labels(),
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:  "configmap-registry-server",
+					Image: image,
+					Args:  []string{"-c", s.GetName(), "-n", s.GetNamespace()},
+					Ports: []v1.ContainerPort{
+						{
+							Name:          "grpc",
+							ContainerPort: 50051,
+						},
+					},
+				},
+			},
+			ServiceAccountName: s.GetName() + "-configmap-server",
+		},
+	}
+	ownerutil.AddOwner(pod, s.CatalogSource, false, false)
+	return pod
+}
+
+func (s *catalogSourceDecorator) ServiceAccount() *v1.ServiceAccount {
+	sa := &v1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.serviceAccountName(),
+			Namespace: s.GetNamespace(),
+		},
+	}
+	ownerutil.AddOwner(sa, s.CatalogSource, false, false)
+	return sa
+}
+
+func (s *catalogSourceDecorator) Role() *rbacv1.Role {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.roleName(),
+			Namespace: s.GetNamespace(),
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				Verbs:         []string{"get"},
+				APIGroups:     []string{""},
+				Resources:     []string{"configmaps"},
+				ResourceNames: []string{s.Spec.ConfigMap},
+			},
+		},
+	}
+	ownerutil.AddOwner(role, s.CatalogSource, false, false)
+	return role
+}
+
+func (s *catalogSourceDecorator) RoleBinding() *rbacv1.RoleBinding {
+	rb := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      s.GetName() + "-server-configmap-reader",
+			Namespace: s.GetNamespace(),
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      s.serviceAccountName(),
+				Namespace: s.GetNamespace(),
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     s.roleName(),
+		},
+	}
+	ownerutil.AddOwner(rb, s.CatalogSource, false, false)
+	return rb
+}
+
+type RegistryReconciler interface {
```

Comment:
Can we potentially not introduce [another class that ends in -er](https://www.yegor256.com/2015/03/09/objects-end-with-er.html)?

Reply:
In general I think naming doesn't matter nearly as much as interface shapes, and names should just be whatever will help a reader the most. So if there's a name that would help you read it better, I would be happy to change it here 😁